### PR TITLE
Login Suggestions

### PR DESCRIPTION
**Pull request by manuartero**
Hi Ines, 

again i took the liberty to fork your solution and play around a little bit. 

This time i've focused on split the functionality: one of the most relevant things to master is knowing when to split the code and when no to. 
So, we have this component , the `<Login />` form. which renders the thing and fetches the API , sending the username and the password to the API, the API would return if the username+pass is ok or not. Depending on the response this component will call the father (`<App />`) or render some red msg.

I'd say , you can extract the API-connection-thing to a different function. It's common to have , next to `components`, a `services` folder, so i created `services/api.js` , here we could - eventually - have a complete menu (API) of methods for fetching or sending data to the API (the service)

***

just to add more ideas, one common thing of this `<Login />` form could be to resolve the token (and the username) from the API back to the parent component. Relax i didn't mention this in the e3xercise, i'm adding this as extra info to you.

So the parent component could recieve on the success method: «ey, this username has this valid token, maybe you can create a **session**? for this fella?»

i added a super super simple example calling [`localStorage.setItem()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) so we are saving in the memory of the browser the login info 👍 

Captura de pantalla 2024-03-18 a las 20 01 25

hope you find this useful.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 function App() {
-  
   const [isLoggedIn, setIsLoggedIn] = useState(false);
 
-  const handleLogin = () => {
+  const handleLogin = ({ username, token }) => {
+    localStorage.setItem("username", username);
+    localStorage.setItem("token", token);
     setIsLoggedIn(true);
   };
```
Comment (manuartero) [inline src/App.jsx:14]: this function is called back when the login is sucessful, this is the way to receive the success info back in the parent component

File: src/services/api.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    }),
+  });
+
+  if (!response.ok) {
+    throw new Error("Wrong Login. Please try again");
+  }
+
+  const { token } = await response.json();
+  return token;
+}
```
Comment (manuartero) [inline src/services/api.js:19]: independent / isolated function

File: src/App.jsx (hunk 1 above)
Comment (xeni09) [reply src/App.jsx:14]: Hey Manu, thanks again, may i ask, probably a stupid question...but... what is the use of this? why do we want to locally storage this?

Comment (xeni09) [reply src/App.jsx:14]: I mean, you mentioned above the line: So the parent component could recieve on the success method: «ey, this username has this valid token, maybe you can create a session? for this fella?»

I guess, its simply that we get back the info so we can use it for anything we need, keep the session logged in or something in this direction?

File: src/services/api.js (hunk 2 above)
Comment (xeni09) [reply src/services/api.js:19]: It makes sense... thanks, these are the type of things I still need to learn from the practical part, the day-to-day experience... and seeing many real projects.

File: src/App.jsx (hunk 1 above)
Comment (manuartero) [reply src/App.jsx:14]: this is just an example but usually, once the login is successful you'll need the user info: the username and the auth token (these tokens hold info inside them) seems reasonable.


1. im a guest (not logged in yet), the home page shows the "guest mode" (not 


Generic cards for a guest (USA based in this website)

Captura de pantalla 2024-03-19 a las 13 03 49

Captura de pantalla 2024-03-19 a las 13 00 16

****

now i log in, the home has the user info avaialble with a lot of customiz

ation, the home page shows MY stuff

➡️ ➡️ ➡️ ➡️ ➡️ ➡️ 

Captura de pantalla 2024-03-19 a las 13 05 23

Captura de pantalla 2024-03-19 a las 13 00 26

***

The user session is a world itself, i wanted just to point you there is a thing named `LocalStorage` where usually you can save these kind of info

Comment (xeni09) [reply src/App.jsx:14]: Got it!! thankssss i will keep trying to understand the whole world of it ^^

**Pull request merged**